tests/test_class.cpp

```diff
@@ -62,7 +62,15 @@ TEST_SUBMODULE(class_, m) {
     };
 
     py::class_<NoConstructor>(m, "NoConstructor")
-        .def_static("new_instance", &NoConstructor::new_instance, "Return an instance");
+        .def_static("new_instance", &NoConstructor::new_instance, "Return an instance")
+        .def_classmethod(
+            "new_instance_uuid",
```

Comment:
Maybe `serial` or `seq` or similar? (It doesn't look like a universally unique identifier.)